### PR TITLE
add PR Gatekeeper

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

replacement of #12599
just after creation it is:
image
after @carmocca approval, it is:
image

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/workflows/ci_pr-gatekeeper.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      [
+        assigned,
+        unassigned,
+        opened,
+        reopened,
+        synchronize,
+        review_requested,
+        review_request_removed
+      ]
+  pull_request_review:
```
Comment (carmocca) [inline .github/workflows/ci_pr-gatekeeper.yml:15]: Is this empty on purpose?

File: .github/approve_config.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+approvals:
+  # check will fail if there is no approval
+  minimum: 1     # optional
+  groups:        # optional
+    - name: GridAI
+      minimum: 1 # optional
```
Comment (carmocca) [inline .github/approve_config.yml:6]:
```suggestion
      minimum: 1
```

File: .github/approve_config.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+approvals:
+  # check will fail if there is no approval
+  minimum: 1     # optional
```
Comment (carmocca) [inline .github/approve_config.yml:3]: This is probably redundant if we specify it below.

**Review by carmocca — APPROVED** [review]
Lets see if it works

File: .github/workflows/ci_pr-gatekeeper.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/ci_pr-gatekeeper.yml:15]: it means it takes all

**Review by rohitgr7 — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Pull request merged**